src/app/services/app-state.service.ts

```diff
@@ -76,11 +76,15 @@ export class AppStateService {
   }
 
 
-  setRecordSelected(cinchyId: number | null, doNotReloadForm: boolean = false): void {
+  setRecordSelected(rowId: number | null, doNotReloadForm: boolean = false): void {
+    this.onRecordSelected$.next({ cinchyId: rowId, doNotReloadForm });
 
-    this._rowId = cinchyId;
-
-    this.onRecordSelected$.next({ cinchyId, doNotReloadForm });
+    if (rowId === null) {
+      this.deleteRowIdInQueryParams();
+    }
+    else {
+      this.updateRowIdInQueryParams(rowId);
+    }
 
     // Update URL with the new ID, if present
```

Comment:
The reason behind creating deleteRowIdInQueryParams and updateRowIdInQueryParams creating separately because they are different. deleteRowIdInQueryParams  calls when we create a new record and updateRowIdInQueryParams when we update a record. If I move this code to updateRowIdInQueryParams then both functions will be tightly coupled.

Reply:
But it's doing the same thing, just in two places. The purpose of the function is to update the rowId parameter, and that can exist in either the applet's URL or as part of the wrapping page's URL when the app is embedded -- or even both. If we want to do so, we therefore need to check both places.

I'm fine if we break the individual parts of that into separate, private functions, but the function `updateRowIdInQueryParams` needs to ensure that it does what it claims to. Everything below the above if statement is performing that role, and so it should be encapsulated in the function.